lib/common/crmcommon_private.h

```diff
@@ -285,4 +284,42 @@ pcmk__ipc_methods_t *pcmk__pacemakerd_api_methods(void);
 G_GNUC_INTERNAL
 extern bool pcmk__is_daemon;
 
+/*!
+ * \brief Check the authenticity of the IPC socket peer process
+ *
+ * If everything goes well, peer's authenticity is verified by the means
+ * of comparing against provided referential UID and GID (either satisfies),
+ * and the result of this check can be deduced from the return value.
+ * As an exception, detected UID of 0 ("root") satisfies arbitrary
+ * provided referential daemon's credentials.
+ *
+ * \param[in]  qb_ipc  libqb client connection if available
+ * \param[in]  sock    IPC related, connected Unix socket to check peer of
+ * \param[in]  refuid  referential UID to check against
+ * \param[in]  refgid  referential GID to check against
+ * \param[out] gotpid  to optionally store obtained PID of the peer
+ *                     (not available on FreeBSD, special value of 1
+ *                     used instead, and the caller is required to
+ *                     special case this value respectively)
+ * \param[out] gotuid  to optionally store obtained UID of the peer
+ * \param[out] gotgid  to optionally store obtained GID of the peer
+ *
+ * \return Standard Pacemaker return code
+ *         ie > 1 if the connection is not authentic 0, if it is
+ *         < 1 for -errno
```

Comment:
For the standard codes, 0 = ok (which would be authentic in this case), >0 is a system errno and <0 is one of the pcmk_rc_* error codes. In this case we probably only care about pcmk_rc_ok = authentic, pcmk_rc_ipc_unauthorized = definitely not authentic, anything else = error (unable to determine).